crates/iota-types/src/move_authenticator.rs

```diff
@@ -69,12 +70,69 @@ impl MoveAuthenticator {
         &self.type_arguments
     }
 
+    pub fn object_to_authenticate(&self) -> &CallArg {
+        &self.object_to_authenticate
+    }
+
+    pub fn object_to_authenticate_components(
+        &self,
+    ) -> IotaResult<(ObjectID, SequenceNumber, Option<ObjectDigest>)> {
```

Comment:
```suggestion
    ) -> IotaResult<(ObjectID, Option<SequenceNumber>, Option<ObjectDigest>)> {
```
The `initial_shared_version` and the object.version() for shared objects are two different versions. So this should return (ObjectId, None, None) when it matches `ObjectArg::SharedObject`.

Reply:
fixed in e2e1b7beb7